fix(aggregate transform): Fix incorrectly dropped events bug in metrics aggregate transform (#25486)
vectordotdev/vector: changelog.d/aggregate_passthrough_ignored_kind.fix.md

The `aggregate` transform now correctly passes through/ignores metrics whose kind is not supported
by the configured mode. Prior to this change these metrics would be silently dropped, contrary to
the officially documented behavior. For example, `absolute` metrics flowing through a `sum`-mode aggregate
transform are now forwarded to the next step in the pipeline unchanged rather than being dropped:

```text
{kind: incremental, type: counter, name: "http.requests", value: 10}  → summed into aggregate
{kind: absolute,    type: gauge,   name: "cpu.usage",     value: 0.83} → previously dropped, now passes through unchanged
{kind: incremental, type: counter, name: "http.requests", value: 5}   → summed into aggregate
```

> If the previous drop behavior was intentional, add a `filter` transform before the aggregate transform to discard the unwanted metric kind.

Reviewer: ```suggestion
If you want to preserve the previous drop behavior, add a `filter` transform before the aggregate transform to discard the unwanted metric kind.
```

> The `aggregate` transform now correctly passes through metrics whose kind is not supported by the configured mode, rather than silently dropping them. For example, `absolute` metrics flowing through a `sum`-mode aggregate are forwarded to the output unchanged without any aggregation & without being dropped

Reviewer: Can we provide an actual example here? And suggest an alternative for users who still want the old behavior e.g. add a filter transform before the aggregate transform.

Author: Updated with an example, let me know if you had something else in mind
